src/store/actionTypes.ts

```diff
@@ -121,145 +127,121 @@ interface UpdateShowResultNameActionPayload {
-
-export interface UpdateFetchingFlowAction {
-    type: Constants.UPDATE_FETCHING_FLOW;
-    payload: FetchingFlowPayload;
-}
-
-export interface UpdateDefinitionAction {
-    type: Constants.UPDATE_DEFINITION;
-    payload: DefinitionPayload;
-}
-
-export interface UpdateNodeDraggingAction {
-    type: Constants.UPDATE_NODE_DRAGGING;
-    payload: NodeDraggingPayload;
-}
-
-export interface UpdateFlowsAction {
-    type: Constants.UPDATE_FLOWS;
-    payload: UpdateFlowsPayload;
-}
-
-export interface UpdateDependenciesAction {
-    type: Constants.UPDATE_DEPENDENCIES;
-    payload: UpdateDependenciesPayload;
-}
-
-export interface UpdatePendingConnectionsAction {
-    type: Constants.UPDATE_PENDING_CONNECTIONS;
-    payload: UpdatePendingConnectionsPayload;
-}
-
-export interface RemovePendingConnectionAction {
-    type: Constants.REMOVE_PENDING_CONNECTION;
-    payload: RemovePendingConnectionPayload;
-}
-
-export interface UpdateComponentsAction {
-    type: Constants.UPDATE_COMPONENTS;
-    payload: UpdateComponentsPayload;
-}
-
-export interface UpdateContactFieldsAction {
-    type: Constants.UPDATE_CONTACT_FIELDS;
-    payload: UpdateContactFieldsPayload;
-}
-
-export interface UpdateGroupsAction {
-    type: Constants.UPDATE_GROUPS;
-    payload: UpdateGroupsPayload;
-}
-
-export interface UpdateResultNamesAction {
-    type: Constants.UPDATE_RESULT_NAMES;
-    payload: UpdateResultNamesPayload;
-}
-
-export interface UpdateNodesAction {
-    type: Constants.UPDATE_NODES;
-    payload: UpdateNodesPayload;
-}
-
-export interface UpdateFreshestNodeAction {
-    type: Constants.UPDATE_FRESHEST_NODE;
-    payload: UpdateFreshestNodePayload;
-}
-
-export interface UpdateNodeEditorOpenAction {
-    type: Constants.UPDATE_NODE_EDITOR_OPEN;
-    payload: UpdateNodeEditorOpenPayload;
-}
-
-export interface UpdateGhostNodeAction {
-    type: Constants.UPDATE_GHOST_NODE;
-    payload: UpdateGhostNodePayload;
-}
-
-export interface UpdateCreateNodePositionAction {
-    type: Constants.UPDATE_CREATE_NODE_POSITION;
-    payload: UpdateCreateNodePositionPayload;
-}
-
-export interface UpdatePendingConnectionAction {
-    type: Constants.UPDATE_PENDING_CONNECTION;
-    payload: UpdatePendingConnectionPayload;
-}
-
-export interface UpdateActionToEditAction {
-    type: Constants.UPDATE_ACTION_TO_EDIT;
-    payload: UpdateActionToEditPayload;
-}
-
-export interface UpdateNodeToEditAction {
-    type: Constants.UPDATE_NODE_TO_EDIT;
-    payload: UpdateNodeToEditPayload;
-}
-
-export interface UpdateLocalizationsAction {
-    type: Constants.UPDATE_LOCALIZATIONS;
-    payload: UpdateLocalizationsPayload;
-}
-
-export interface UpdateDragGroupAction {
-    type: Constants.UPDATE_DRAG_GROUP;
-    payload: UpdateDragGroupPayload;
-}
-
-export interface UpdateTypeConfigAction {
-    type: Constants.UPDATE_TYPE_CONFIG;
-    payload: UpdateTypeConfigPayload;
-}
+export type UpdateTranslatingAction = DuxAction<Constants.UPDATE_TRANSLATING, TranslatingPayload>;
```

Comment:
generics are blowing my mind right now; can see a ton of uses for these elsewhere in the repo...action generic, router generic, etc.